src/code/utilities/authoring-utils.js

```diff
@@ -0,0 +1,51 @@
+export default class AuthoringUtils {
+  static getChallengeDefinition(authoring, routeSpec) {
+    const challengeId = this.getChallengeId(authoring, routeSpec),
+          challenges = authoring.challenges;
+    return challengeId && challenges ? authoring.challenges[challengeId] : null;
+  }
+
+  static getChallengeId(authoring, routeSpec) {
+    const levels = authoring.application.levels,
+          missions = levels ? levels[routeSpec.level].missions : null,
+          challenges = missions ? missions[routeSpec.mission].challenges : null,
+          challenge = challenges ? challenges[routeSpec.challenge] : null;
+    return challenge ? challenge.id : null;
```

Comment:
If you just replaced the last one it would return null anyway. If you replaced all of the ternaries, it could potentially return undefined if levels were undefined initially.